src/trustedtypes.js

```diff
@@ -289,16 +349,35 @@ export const TrustedTypes = (function() {
    * @todo Figure out if the return value (and the builder) can be typed.
    */
   function createPolicy(name, builder) {
+    // TODO(msamuel): could we also allow symbol names for non-exposed policies?
+    // That would simplify a lot of collision avoidance stuff in client code.
+    // For example,
+    // function toPolicyName(name) {
+    //   if ('symbol' === typeof name) {
+    //     if (getPrototypeOf(name) !== Symbol.prototype) {
+    //       throw new Error('Symbols from other Realms cannot be policy names')
+    //     }
+    //     return name;
+    //   }
+    //   return '' + name;  // Require a string.
+    // }
     const pName = '' + name; // Assert it's a string
 
     if (policyNames.includes(pName)) {
       throw new Error('Policy ' + pName + ' exists');
     }
+    // Register the name early so that if builder unwisely calls
+    // across protection domains to code that reenters this function,
+    // builder's author still has rights to the name.
+    policyNames.push(pName);
 
-    let innerPolicy = Object.assign({}, initialBuilder);
+    // TODO(msamuel): can the innerPolicy make do with a null prototype?
```

Comment:
Yes, I think we can. If this turns out to be a problem with actual real-life policies, we can relax this.

Reply:
Done